multimodal/src/autogluon/multimodal/optimization/lit_distiller.py

```diff
@@ -154,15 +171,34 @@ def __init__(
         self.temperature = temperature
         self.hard_label_weight = hard_label_weight
         self.soft_label_weight = soft_label_weight
+        self.intermediate_loss_weight = intermediate_loss_weight
+        self.rkd_distance_loss_weight = rkd_distance_loss_weight
+        self.rkd_angle_loss_weight = rkd_angle_loss_weight
         self.hard_label_loss_func = hard_label_loss_func
         self.soft_label_loss_func = soft_label_loss_func
+        self.intermediate_loss_func = intermediate_loss_func
         if isinstance(validation_metric, BaseAggregator) and custom_metric_func is None:
             raise ValueError(
                 f"validation_metric {validation_metric} is an aggregation metric,"
                 "which must be used with a customized metric function."
             )
         self.custom_metric_func = custom_metric_func
 
+        # Adapt student's intermediate feature to teacher's
+        if isinstance(self.teacher_model, MultimodalFusionMLP) or isinstance(self.teacher_model, MultimodalFusionTransformer):
+            pass # skip fusion model for now
+        else:
+            teacher_model_dim = self.teacher_model.out_features
+            student_model_dim = self.student_model.out_features
+            self.intermediate_adaptor = (
+                nn.Linear(teacher_model_dim, student_model_dim)
+                if teacher_model_dim != student_model_dim
+                else nn.Identity()
+            )
+
+            self.rkd_distance = RKDDistance()
+            self.rkd_angle = RKDAngle()
```

Comment:
We'd better create the adaptor, rkd_distance_func and rkd_angle_func in `_setup_distillation()`, which set up all the required modules for distillation. This file `lit_distiller.py` is mainly to define optimization loops.

Reply:
Done. Moved the creation of adaptor and rkd_loss to _setup_distillation().